sources/osgUtil/PolytopeIntersectFunctor.js

```diff
@@ -0,0 +1,301 @@
+'use strict';
+
+var vec3 = require( 'osg/glMatrix' ).vec3;
+var PrimitiveFunctor = require( 'osg/PrimitiveFunctor' );
+var Plane = require( 'osg/Plane' );
+var intersectionEnums = require( 'osgUtil/intersectionEnums' );
+
+var PolytopeIntersection = function () {
+    this._intersectionPoints = [];
+    this._primitiveIndex = undefined;
+    this._distance = 0;
+    this._maxDistance = 0;
+    this._nodePath = undefined;
+    this._drawable = undefined;
+    this._matrix = undefined;
+    this._localIntersectionPoint = undefined;
+    this._numIntersectionPoints = 0;
+};
+
+// Settings are needed.
+var PolytopeIntersectFunctor = function ( settings ) {
+    this._src = [];
+    this._dest = [];
+    this._settings = settings;
+    this._primitiveIndex = 0;
+    this._hit = false;
+    this._maxNumIntersectionsPoints = 6;
+    this._primitiveMask = intersectionEnums.ALL_PRIMITIVES;
+};
+
+PolytopeIntersectFunctor.prototype = {
+
+    enter: function ( bbox ) {
+        if ( this._settings._polytopeIntersector.getPolytope().containsBoundingBox( bbox ) ) {
+            this._settings._polytopeIntersector.getPolytope().pushCurrentMask();
+            return true;
+        } else {
+            return false;
+        }
+    },
+
+    leave: function () {
+        this._settings._polytopeIntersector.getPolytope().popCurrentMask();
+    },
+
+    addIntersection: function () {
+
+        var uniq = function ( a ) {
```

Comment:
Does not seem efficient, but maybe it would need an id on src to check directly in a {}
but maybe we can't add id to src or performance on this part is not critical